src/services/aave/aave.service.ts

```diff
@@ -21,54 +22,73 @@ export default class AaveService {
     prices: TokenPrices,
     currency: FiatCurrency
   ) {
+    const { mainTokens, wrappedTokens, linearPoolTokensMap } = pool;
+    const wrappedTokenBalances = wrappedTokens?.map(
+      token => linearPoolTokensMap?.[token].balance || ''
+    );
+    const hasAllWrappedTokenBalances =
+      wrappedTokenBalances && wrappedTokenBalances.every(balance => !!balance);
+
+    if (!mainTokens || !wrappedTokens || !hasAllWrappedTokenBalances)
+      return { total: '0', breakdown: {} };
+
     let total = bnum(0);
-    const tokenBreakdown = {};
-
-    if (pool.mainTokens != null && pool.wrappedTokens != null) {
-      const reserves = await this.subgraphService.reserves.get({
-        where: {
-          underlyingAsset_in: pool.mainTokens,
-          isActive: true,
-          aEmissionPerSecond_gt: 0
-        }
-      });
-
-      reserves.forEach(reserve => {
-        const supplyAPR = bnum(reserve.supplyAPR);
-
-        if (
-          pool.mainTokens != null &&
-          pool.wrappedTokens != null &&
-          supplyAPR.gt(0)
-        ) {
-          const tokenIndex = pool.mainTokens.findIndex(
-            token => token === getAddress(reserve.underlyingAsset)
+    const breakdown: Record<string, string> = {};
+
+    const reserves = await this.subgraphService.reserves.get({
+      where: {
+        underlyingAsset_in: mainTokens,
+        isActive: true
+      }
+    });
```

Comment:
hey, noticed you had to filter the reserves list below, but I believe if Aave stops LM we might pick the wrong one. I suggest we add a clause to this query `aTokens_in: unwrappedTokens` - IIUC we have access to unwrappedTokens list in Pool object which is the list with aDAI, aUSDC, aUSDT, right? With that, reserves will be a 3-items list and you don't have to worry about filtering.

Reply:
Hey, you might be right, could you implement a fix and PR off of this branch?